src/main/java/com/imsweb/x12/reader/X12Reader.java

```diff
@@ -562,13 +562,19 @@ private boolean validateLines(List<String> segments, String loopId, Separators s
         int[] segmentCounter = new int[format.size()];
         boolean lineMatchesFormat = false;
 
+        String previousPos = null;
         for (String segment : segments) {
             int i = 0;
             for (SegmentDefinition segmentConf : format) {
                 String[] tokens = segment.split(Pattern.quote(separators.getElement().toString()));
                 if (tokens[0].equals(segmentConf.getXid()) && codesValidated(tokens, segmentConf)) {
+                    String currentPos = segmentConf.getPos();
+                    if(previousPos != null && Integer.parseInt(previousPos) > Integer.parseInt(currentPos))
```

Comment:
Is your code formatter set up correctly?  There is no space after "if".

Reply:
Sorry, for some reason when I opened up that project in a separate IntelliJ window the Reformat Code button was not checked like it normally is.